src/orquestra/integrations/qiskit/backend/backend.py

```diff
@@ -374,15 +377,19 @@ def _apply_readout_correction(
             num_qubits = len(key)
             break
 
-        if active_qubits is None:
-            active_qubits = list(range(num_qubits))
+        if virtual_to_physical_qubits_dict is None:
+            virtual_qubits = list(range(num_qubits))
+            physical_qubits = list(range(num_qubits))
         else:
-            active_qubits.sort()
+            virtual_qubits = list(virtual_to_physical_qubits_dict.keys())
+            physical_qubits = list(virtual_to_physical_qubits_dict.values())
+            virtual_qubits.sort()
+            physical_qubits.sort()
```

Comment:
Do not sort the `physical_qubits`. 
Suppose, we have a `v2p = {0: 4, 1: 3, 3: 0, 4: 2, 2: 1}`. The `physical_qubits` should be `[4, 3, 1, 0, 2]` not `[0, 1, 2, 3, 4]`. Instead of sorting, make sure that the value @ key-th index of `physical_qubits` corresponds to the (key, value) pair in the `v2p`.

Reply:
I think it is sorted to prevent having multiple keys for the same filter in `readout_correction_filters`. Does the order of the qubits affect the calibration? If not, I think it is okay to keep `physical_qubits` sorted.

Reply:
Yes, the order is important. It affects the calibration matrix. :)

Reply:
We sort `virtual_qubits` so that the counts we get from `marginal_counts(counts, virtual_qubits)` are in the correct order, so I think we should do 
```
physical_qubits = [
    virtual_to_physical_qubits_dict[virtual_qubit]
    for virtual_qubit in virtual_qubits
]
```
so that the i-th `physical_qubit` corresponds to the i-th `virtual_qubit`

Reply:
Looks good now!

Very minor suggestion:

The following 2 lines
```
virtual_qubits = list(virtual_to_physical_qubits_dict.keys())
virtual_qubits.sort()
```

can be replaced by a single line

```
virtual_qubits = sorted(virtual_to_physical_qubits_dict.keys())
```